pkg/transform/oauth/ldap.go

```diff
@@ -9,21 +9,30 @@ import (
 // IdentityProviderLDAP is a LDAP specific identity provider
 type IdentityProviderLDAP struct {
 	identityProviderCommon `yaml:",inline"`
-	LDAP                   struct {
-		Attributes struct {
-			ID                []string `yaml:"id"`
-			Email             []string `yaml:"email"`
-			Name              []string `yaml:"name"`
-			PreferredUsername []string `yaml:"preferredUsername"`
-		} `yaml:"attributes"`
-		BindDN       string `yaml:"bindDN"`
-		BindPassword string `yaml:"bindPassword"`
-		CA           struct {
-			Name string `yaml:"name"`
-		} `yaml:"ca"`
-		Insecure bool   `yaml:"insecure"`
-		URL      string `yaml:"url"`
-	} `yaml:"ldap"`
+	LDAP                   LDAP `yaml:"ldap"`
+}
+
+// LDAP provider specific data
+type LDAP struct {
+	Attributes   LDAPAttributes `yaml:"attributes"`
+	BindDN       string         `yaml:"bindDN"`
+	BindPassword string         `yaml:"bindPassword"`
+	CA           LDAPCA         `yaml:"ca"`
+	Insecure     bool           `yaml:"insecure"`
+	URL          string         `yaml:"url"`
+}
+
+// LDAPCA is the CA cert for LDAP provider
+type LDAPCA struct {
+	Name string `yaml:"name"`
+}
```

Comment:
Use the common `CA` or `CertificateAuthority` struct from the new `types.go` file in oauth package.